### ISSUE TITLE
Full BTRFS support for Fedora out of the box

### ISSUE THREAD

**Issue by ShrirajHegde** [teejee2008/timeshift#764]
**Describe the solution you'd like**
Support for Fedora BTRFS partitioning scheme along with support for fedora boot partition (ext4)

(since timeshift works really really well with BTRFS and Fedora community has huge BTRFS users since fedora33 and above use BTRFS by default)

**Describe alternatives you've considered**
1. Using ext4 backup - It is slow both for backup and restore, requires a different partition, and restores wear SSD a lot not to mention the inconvenience of reboot
2. Renaming BTRFS subvolumes after install / setting up right ubuntu-like subvolumes right at installation time (which still leaves out boot partition which has to be backed up manually)

**Additional context**
It is a great time for fedora since it is being user friendly like Ubuntu while providing the latest technology, kernel, software, and features to absolute beginners, timeshift is will be a perfect target for such demographic (especially singe timeshift-gtk provides a very intuitive and easy to use GUI)

**Comment by jwatt** [comment]
The partition layout issue in possibly covered by or at least related to issue #241, as is issue #665.

**Comment by ShrirajHegde** [comment]
> The partition layout issue in possibly covered by or at least related to issue #241, as is issue #665.

yes but boot partition being left out is a huge issue because a problem in boot partition is one of the top issues

It would be great if timeshift offers preset for Fedora, OpenSUSE and manually selecting subvolumes for custom subvolume scheme, along with rsync,btrfs, and a new  **HYBRID** mode where the boot can be backed up with rsync while BTRFS can use snapshots. This is kind of the ideal situation since GRUB can only read BTRFS, doesn't write to it.

These altogether will add full flexibility that any distro can use Timeshift

**Comment by imrj** [comment]
i didnt know about this issue and had been using timeshift to backup my fedora 34 system.....well I lost my SSD and now i cant restore my fedora timeshift snaps to a new SSD.......hopefully there is a fix soon, I hate to reinstall everything from scratch

**Comment by kheshav-coder** [comment]
> **Describe the solution you'd like**
> Support for Fedora BTRFS partitioning scheme along with support for fedora boot partition (ext4)
> 
> (since timeshift works really really well with BTRFS and Fedora community has huge BTRFS users since fedora33 and above use BTRFS by default)
> 
> **Describe alternatives you've considered**
> 
>     1. Using ext4 backup - It is slow both for backup and restore, requires a different partition, and restores wear SSD a lot not to mention the inconvenience of reboot
> 
>     2. Renaming BTRFS subvolumes after install / setting up right ubuntu-like subvolumes right at installation time (which still leaves out boot partition which has to be backed up manually)
> 
> 
> **Additional context**
> It is a great time for fedora since it is being user friendly like Ubuntu while providing the latest technology, kernel, software, and features to absolute beginners, timeshift is will be a perfect target for such demographic (especially singe timeshift-gtk provides a very intuitive and easy to use GUI)

I really appreciate your request, I too want the same thing to be done. as said there are a ton of fedora users who would love to use timeshift out of the box with boot partition included. please make the update as soon as possible.

**Comment by teejee2008** [comment]
Timeshift was originally written for Ubuntu-based distributions. Adding support for other Fedora, SUSE, etc will require some re-work. If someone is interested in implementing this please go ahead.

**Comment by ShrirajHegde** [comment]
> Timeshift was originally written for Ubuntu-based distributions. Adding support for other Fedora, SUSE, etc will require some re-work. If someone is interested in implementing this please go ahead.

The quick fix would be to let users select names of subvolumes. Fedora has similar separate subvolumes for `home` and `/`except with different names. (`home`, `root` instead of `@` and `@home` in ubuntu)

As mentioned here:
https://web.archive.org/web/20210127021028/https://mutschler.eu/linux/install-guides/fedora-btrfs/#create-btrfs-subvolumes--and-home

This still leaves out `/boot` which is on a separate ext4 partition. So including this would need rsync and btrfs-snapshots at the same time. If this is implemented, fedora might as well include timeshift out of the box since btrfs snapshots are very very convenient on fedora.

But, usually boot partition restoration isn't necessary (Timeshift is working fine for me after just renaming subvolumes). People who desperately want the boot partition to be snapshotted also can move the boot partition back to root partition and update grub configs.
